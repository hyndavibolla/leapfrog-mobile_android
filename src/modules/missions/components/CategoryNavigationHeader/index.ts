export { default as CategoryNavigationHeader } from './CategoryNavigationHeader';
