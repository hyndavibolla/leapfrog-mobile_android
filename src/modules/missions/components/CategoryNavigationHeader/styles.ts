import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE, colorWithOpacity } from '_constants';

export const styles = {
  shadow: css`
    shadow-color: ${COLOR.BLACK};
    shadow-offset: 0px 4px;
    shadow-opacity: 0.1;
    elevation: 3;
  `,
  body: css`
    width: 100%;
    align-items: center;
    justify-content: center;
    flex-direction: row;
  `,
  closeBtn: css`
    justify-content: center;
    align-items: center;
    width: 40px;
    height: 40px;
  `,
  container: css`
    flex-direction: row;
    padding-horizontal: 20px;
  `,
  safeArea: css`
    background-color: ${colorWithOpacity(COLOR.BLACK, 35)};
    height: 100%;
    justify-content: center;
  `,
  backgroundContainer: css`
    background-color: ${COLOR.WHITE};
    height: 108px;
    overflow: hidden;
  `,
  title: css`
    color: ${COLOR.WHITE};
    font-size: ${FONT_SIZE.REGULAR};
    font-family: ${FONT_FAMILY.HEAVY};
    text-align: center;
    justify-content: center;
  `,
  searchBarContainer: css`
    flex-grow: 2;
    padding-left: 16px;
  `
};
