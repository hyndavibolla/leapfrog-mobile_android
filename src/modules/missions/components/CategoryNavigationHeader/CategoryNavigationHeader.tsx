import React, { memo } from 'react';
import { StackHeaderProps } from '@react-navigation/stack';
import { SafeAreaView } from 'react-native-safe-area-context';
import { View, StatusBar } from 'react-native';

import { Icon } from '_commons/components/atoms/Icon';
import { Button } from '_components/Button';
import { ImageBackgroundWithFallback } from '_components/ImageBackgroundWithFallback';
import { COLOR, CONTAINER_STYLE, FONT_SIZE, ICON } from '_constants';
import { useTestingHelper } from '_utils/useTestingHelper';
import { styles } from './styles';

export interface Props extends StackHeaderProps {
  backgroundSrc: string;
  searchBar?: React.ReactNode;
}

export const CategoryNavigationHeader = ({ backgroundSrc, navigation, searchBar }: Props) => {
  const { getTestIdProps } = useTestingHelper('category-navigation-header');
  return (
    <ImageBackgroundWithFallback
      containerStyle={styles.backgroundContainer}
      source={{ uri: backgroundSrc }}
      fallbackSource={require('_assets/shared/categoryBackgroundFallback.png')}
    >
      <StatusBar translucent backgroundColor={COLOR.TRANSPARENT} barStyle="light-content" />
      <SafeAreaView style={[styles.safeArea]} edges={['top']}>
        <View style={styles.body}>
          <View style={styles.container}>
            <Button onPress={navigation?.goBack} containerColor={COLOR.WHITE} innerContainerStyle={[CONTAINER_STYLE.shadow, styles.closeBtn]}>
              <Icon name={ICON.ARROW_LEFT} color={COLOR.BLACK} size={FONT_SIZE.SMALLER} />
            </Button>
            {searchBar && (
              <View {...getTestIdProps('search-bar')} style={styles.searchBarContainer}>
                {searchBar}
              </View>
            )}
          </View>
        </View>
      </SafeAreaView>
    </ImageBackgroundWithFallback>
  );
};

export default memo(CategoryNavigationHeader);
