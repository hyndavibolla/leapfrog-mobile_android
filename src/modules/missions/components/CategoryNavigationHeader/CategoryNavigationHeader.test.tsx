import React from 'react';
import { render } from '@testing-library/react-native';
import { act } from 'react-test-renderer';

import { getMissionCategory_1 } from '_test_utils/entities';
import { wait } from '_utils/wait';
import CategoryNavigationHeader, { Props } from './CategoryNavigationHeader';

jest.mock('@react-navigation/native', () => ({
  __esModule: true,
  useNavigation: () => ({ navigate: jest.fn() })
}));

describe('CategoryNavigationHeader', () => {
  let props: Props;
  beforeEach(() => {
    const category = getMissionCategory_1();
    props = {
      scene: { route: { name: 'test name' } },
      navigation: { goBack: jest.fn(), navigate: jest.fn() },
      backgroundSrc: category.lifestyleUrl
    } as any;
  });

  it('should render', () => {
    const { toJSON } = render(<CategoryNavigationHeader {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with a bottom shadow', () => {
    (props.scene.route as any).params = { bottomShadow: true };
    const { toJSON } = render(<CategoryNavigationHeader {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should show a search bar', async () => {
    props.searchBar = <></>;
    const { queryByTestId } = render(<CategoryNavigationHeader {...props} />);
    await act(() => wait(0));
    expect(queryByTestId('category-navigation-header-search-bar')).toBeTruthy();
  });
});
