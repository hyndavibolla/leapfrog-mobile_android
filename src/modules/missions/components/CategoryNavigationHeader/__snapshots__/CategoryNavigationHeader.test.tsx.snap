// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryNavigationHeader should render 1`] = `
<View
  accessibilityLabel="image-background-container"
  accessible={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "height": 108,
      "overflow": "hidden",
    }
  }
  testID="image-background-container"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 108,
            "width": undefined,
          },
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      accessibilityLabel="image-background-image"
      accessible={true}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Apparel_&_Accessories-_AGGGFVI9CG",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="image-background-image"
    />
    <RNCSafeAreaView
      edges={
        Array [
          "top",
        ]
      }
      style={
        Array [
          Object {
            "backgroundColor": "#262C2F35",
            "height": "100%",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0066CC",
                    "borderBottomLeftRadius": 30,
                    "borderBottomRightRadius": 30,
                    "borderTopLeftRadius": 30,
                    "borderTopRightRadius": 30,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "elevation": 3,
                      "shadowColor": "#262C2F",
                      "shadowOffset": Object {
                        "height": 4,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1,
                    },
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "width": 40,
                    },
                  ],
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                ]
              }
            >
              <View
                style={false}
              >
                <Text
                  accessibilityLabel="icon-text"
                  accessible={true}
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#262C2F",
                        "fontSize": 16,
                      },
                      undefined,
                      Object {
                        "fontFamily": "sywmax",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="icon-text"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;

exports[`CategoryNavigationHeader should render with a bottom shadow 1`] = `
<View
  accessibilityLabel="image-background-container"
  accessible={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "height": 108,
      "overflow": "hidden",
    }
  }
  testID="image-background-container"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 108,
            "width": undefined,
          },
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      accessibilityLabel="image-background-image"
      accessible={true}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Apparel_&_Accessories-_AGGGFVI9CG",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="image-background-image"
    />
    <RNCSafeAreaView
      edges={
        Array [
          "top",
        ]
      }
      style={
        Array [
          Object {
            "backgroundColor": "#262C2F35",
            "height": "100%",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0066CC",
                    "borderBottomLeftRadius": 30,
                    "borderBottomRightRadius": 30,
                    "borderTopLeftRadius": 30,
                    "borderTopRightRadius": 30,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "elevation": 3,
                      "shadowColor": "#262C2F",
                      "shadowOffset": Object {
                        "height": 4,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1,
                    },
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "width": 40,
                    },
                  ],
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                ]
              }
            >
              <View
                style={false}
              >
                <Text
                  accessibilityLabel="icon-text"
                  accessible={true}
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#262C2F",
                        "fontSize": 16,
                      },
                      undefined,
                      Object {
                        "fontFamily": "sywmax",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="icon-text"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</View>
`;
